### PR TITLE
[Merged by Bors] - feat: port CategoryTheory.Subobject.Comma

### PR DESCRIPTION
**Pull request by mo271**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by mo271** [comment]
Only thing missing:
the `map_rel_iff'` in both `quotientEquiv` and `subobjectEquiv`

File: Mathlib/CategoryTheory/Subobject/Comma.lean [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+Our computation requires that `C` has all limits and `T` preserves all limits. Furthermore, we
+require that the morphisms of `C` and `D` are in the same universe. It is possible that both of
+these requirements can be relaxed by refining the results about limits in comma categories.
+
+We also provide the dual results. As usual, we use `Subobject (op A)` for the quotient objects of
+`A`.
+
+-/
+-- porting note: the main declaration `StructuredArrow.equivSubtype` as claimed above is not there,
+-- perhaps either `subobjectEquiv` or `quotientEquiv` is meant?!
```
Comment (semorrison) [inline Mathlib/CategoryTheory/Subobject/Comma.lean:36]:
```suggestion
## Main declarations
* `StructuredArrow.subobjectEquiv `: the order-equivalence between `Subobject A` and a subtype of
  `Subobject A.right`.

## Implementation notes
Our computation requires that `C` has all limits and `T` preserves all limits. Furthermore, we
require that the morphisms of `C` and `D` are in the same universe. It is possible that both of
these requirements can be relaxed by refining the results about limits in comma categories.

We also provide the dual results. As usual, we use `Subobject (op A)` for the quotient objects of
`A`.

-/
```

File: Mathlib/CategoryTheory/Subobject/Comma.lean [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+/-!
+# Subobjects in the category of structured arrows
+
+We compute the subobjects of an object `A` in the category `StructuredArrow S T` for `T : C ⥤ D`
+and `S : D` as a subtype of the subobjects of `A.right`. We deduce that `StructuredArrow S T` is
+well-powered if `C` is.
+
+## Main declarations
+* `StructuredArrow.equivSubtype`: the order-equivalence between `Subobject A` and a subtype of
+  `Subobject A.right`.
```
Comment (semorrison) [inline Mathlib/CategoryTheory/Subobject/Comma.lean:24]:
```suggestion
* `StructuredArrow.subobjectEquiv `: the order-equivalence between `Subobject A` and a subtype of
  `Subobject A.right`.
```

File: Mathlib/CategoryTheory/Subobject/Comma.lean (hunk 1 above)
Comment (semorrison) [inline Mathlib/CategoryTheory/Subobject/Comma.lean:36]:
```suggestion
```

**Comment by semorrison** [comment]
bors d+

**Pull request closed without merging**